feat(themes): theme doc updates (#15315)
gatsbyjs/gatsby: docs/docs/themes/getting-started.md

---

> title: Using a Gatsby theme

Reviewer: ```suggestion
title: Using a Gatsby Theme
```

Should we change the filename to match the new title (typically what we recommend for SEO)? We might also want a redirect (could check analytics)

Author: Yeah, should change the filename. No reason not to add a redirect, just in case it's linked somewhere.

---

> To get started using themes, we'll use a starter that installs the official gatsby blog theme.

Reviewer: ```suggestion
To get started using themes, you can use a starter that installs the official Gatsby blog theme.
```
Rephrasing slightly to follow the [Gatsby Style Guide](https://www.gatsbyjs.org/contributing/gatsby-style-guide/#use-you-as-the-pronoun).

> A **regular Gatsby starter** creates a new Gatsby site that is preconfigured for a particular use case. The resulting site effectively forks off the starter — after it’s created, the site maintains no connection to the starter.

Reviewer: An example would help illustrate this. Can we recommend a regular Gatsby starter here, like https://github.com/gatsbyjs/gatsby-starter-blog?

A **Gatsby theme starter** creates a new site that installs one or more Gatsby themes. When a starter installs a theme, it maintains the connection to that theme as a standalone npm package.

> ```shell

Reviewer: ```suggestion
Installing the Gatsby blog theme starter is the same process as a regular Gatsby starter:

```shell
```

gatsby new my-blog https://github.com/gatsbyjs/gatsby-starter-blog-theme
```

## What does a theme starter do?

> The starter for the official gatsby blog theme does the following:

Reviewer: ```suggestion
The starter for the official Gatsby blog theme does the following:
```

### 1. The starter installs the theme and configures it

> When you use a starter that's built with a theme, you will see that you're initially presented with a lighter weight `gatsby-config.js`.

Reviewer: ```suggestion
When you use a starter that's built with a theme, you will often see that you're initially presented with a lighter weight `gatsby-config.js`. Themes start doing their magic when installed via the `plugins` array:
```

I think it's important to not set the expectation that *every* theme starter would be lighter weight. Also, we should probably point out the plugins array as notable (feel free to reword it)

I'm an example post!
```

## Updating a Theme

In order to update the theme to pull in the latest updates you can update the `gatsby-theme-blog` version in your site's `package.json`.

> You can do this by running the install of the theme package again: `npm install --save gatsby-theme-blog`.

Reviewer: I think this doc would benefit from a recommendation of next steps from a content perspective. What would someone do next with what they have set up?
